### ISSUE TITLE
Errors while freeing memory

### ISSUE THREAD

**Issue by ghost** [robovm/robovm#355]
(this is an edited issue description as the previous report was overly broad -- covered more than one core issue -- and without a test case)

Repeatedly creating CGImage, CGDataProvider and NSData objects crashes the app.

If the instances **are not** explicitly freed the test case crashes eventually with an error like this:

```
TestAppDelegate(920,0x3bba418c) malloc: *** mach_vm_map(size=1048576) failed (error code=3)
*** error: can't allocate region securely
*** set a breakpoint in malloc_error_break to debug
```

If the instances **are** explicitly disposed the app crashes much earlier and with a number of different errors, most commonly:

```
TestAppDelegate(901,0x3bba418c) malloc: *** error for object 0x16e46004: incorrect checksum for freed object - object was probably modified after being freed.
*** set a breakpoint in malloc_error_break to debug
```

or

```
TestAppDelegate(981,0x40103000) malloc: *** error for object 0x14e39c30: double free
*** set a breakpoint in malloc_error_break to debug
```

To enable or disable explicit freeing please flip the switch at line 53 (WITH_DISPOSE).

In short, whether the objects are freed explicitly or not the app crashes either way, just sooner or later.

It is of course possible I'm not freeing the objects properly but even if that was the case I believe that .dispose() or .close() API calls should not crash the app.

Test case: https://github.com/Gilead/RoboVMTests/blob/master/ImageMemoryIssues/src/ImageMemoryIssues/TestAppDelegate.java
This is a complete project with a runnable script to launch the app.
